clang/test/CodeGenSYCL/hier_par.cpp

```diff
@@ -32,12 +31,11 @@ void foo() {
 
     cgh.parallel_for_work_group<class hpar_hw>(
         range<1>(1), range<1>(1), [=](group<1> g) {
-// CHECK: @[[SHADOW:[a-zA-Z0-9]+]] = internal unnamed_addr addrspace(3) global %[[GROUP_CLASS:"[^"]+"]] undef, align [[ALIGN:[0-9]+]]
-// CHECK: define {{.*}} spir_func void @{{"[^"]+"}}({{[^,]+}}, %[[GROUP_CLASS]]* byval(%[[GROUP_CLASS]]) align {{[0-9]+}} %[[GROUP_OBJ:[A-Za-z_0-9]+]]) {{.*}}!work_group_scope{{.*}} {
-// CHECK-NOT: {{^[ \t]*define}}
-// CHECK: %[[TMP:[A-Za-z_0-9]+]] = bitcast %[[GROUP_CLASS]] addrspace(3)* @[[SHADOW]] to i8 addrspace(3)*
-// CHECK:  %[[OBJ:[A-Za-z_0-9]+]] = bitcast %[[GROUP_CLASS]]* %[[GROUP_OBJ]] to i8*
-// CHECK:  call void @llvm.memcpy.p3i8.p0i8.i64(i8 addrspace(3)* align [[ALIGN]] %[[TMP]], {{[^,]+}} %[[OBJ]], {{[^)]+}})
+          // CHECK: @[[SHADOW:[a-zA-Z0-9]+]] = internal unnamed_addr addrspace(3) global %[[GROUP_CLASS:"[^"]+"]] undef, align [[ALIGN:[0-9]+]]
```

Comment:
Is LLVM transformation always uses `addrspace(3)` for "local" memory? I think it's supposed to be target depended.
If so, I think we should explicitly specify the target device in the RUN command to make sure that "local" memory is represented as `addrspace(3)`.

Why do we care about `align` attribute?
```suggestion
          // CHECK: @[[SHADOW:[a-zA-Z0-9]+]] = internal unnamed_addr addrspace(3) global %[[GROUP_CLASS:"[^"]+"]] undef
```

Reply:
Good concern, thx, I think I can create different check labels for different targets to address this.

Reply:
I'm fine if you add a TODO and we address this later.